airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -633,36 +647,62 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
 
 
 class BulkIncrementalSalesforceStream(BulkSalesforceStream, IncrementalRestSalesforceStream):
+    STREAM_SLICE_STEP = 120
+
     def next_page_token(self, last_record: Mapping[str, Any]) -> Optional[Mapping[str, Any]]:
         if self.name not in UNSUPPORTED_FILTERING_STREAMS:
-            page_token: str = last_record[self.cursor_field]
-            res = {"next_token": page_token}
-            # use primary key as additional filtering param, if cursor_field is not increased from previous page
-            if self.primary_key and self.prev_start_date == page_token:
-                res["primary_key"] = last_record[self.primary_key]
-            return res
+            return {
+                "next_token": last_record[self.cursor_field],
+                "primary_key": last_record.get(self.primary_key)
+            }
         return None
 
+    def stream_slices(
```

Comment:
@roman-yermilov-gl we want this logic to apply to both IncrementalRestSalesforceStream and BulkIncrementalSalesforceStream, so it should be moved up. That should also allow for cleanup of slice-related logic that was added to `request_params`, since much of it is duplicated between IncrementalRestSalesforceStream and BulkIncrementalSalesforceStream.

Can you also add unit tests for the IncrementalRestSalesforceStream case?

Reply:
Hey @roman-yermilov-gl Just wanted to circle back here. If we address these changes we should be able to merge this in :)

Reply:
@girarda 
Moved slicer to `IncrementalRestSalesforceStream` so all the incremental streams use slicing now

Reply:
@bnchrch 
What should I do?)